widget/embedded/src/store/quote.ts

```diff
@@ -38,21 +41,30 @@ export interface QuoteState {
   outputAmount: BigNumber | null;
   outputUsdValue: BigNumber | null;
   fromToken: Token | null;
+  sortStrategy: SortStrategy;
   toToken: Token | null;
   quoteWalletsConfirmed: boolean;
   selectedWallets: Wallet[];
   quoteWarningsConfirmed: boolean;
+  refetchQuote: boolean;
+  selectedTag: RouteTag | null;
   resetQuote: () => void;
   resetToBlockchain: () => void;
   resetFromBlockchain: () => void;
   setFromBlockchain: (chain: BlockchainMeta | null) => void;
   setToBlockchain: (chian: BlockchainMeta | null) => void;
-
+  setSortStrategy: (sortStrategy: SortStrategy) => void;
+  setRefetchQuote: (refetchQuote: boolean) => void;
   setFromToken: (params: SetTokenParams) => void;
   setToToken: (params: SetTokenParams) => void;
   setInputAmount: (amount: string) => void;
-  quote: BestRouteResponse | null;
-  setQuote: (quote: BestRouteResponse | null) => void;
+  selectedQuote: MultiRouteSimulationResult | BestRouteResponse | null;
```

Comment:
Most of the time, you can simplify this union type to a type that has common properties between "MultiRouteSimulationResult" and "BestRouteResponse." This type should work for all the functions already in place and help do away with type narrowing at different points. Something like this:
```ts
type Quote = { swaps: SwapResult[], requestId: string , ...other common properties}
```

Reply:
fixed